src/FiniteVolumeMLMC.cpp

```diff
@@ -73,80 +124,109 @@ FiniteVolumeMLMC::FiniteVolumeMLMC(MPI_Comm comm,
 /// this implementation is sloppy
 void FiniteVolumeMLMC::RescaleFineCoefficient(const mfem::Vector& coeff)
 {
-    mfem::Vector temp(coeff);
-    for (int i = 0; i < coeff.Size(); ++i)
-        temp(i) = coeff(i) * weight_(i);
-    GetFineMatrix().SetMFromWeightVector(temp);
+    if (!hybridization_)
+    {
+        GetFineMatrix().UpdateM(coeff);
+        ForceMakeFineSolver();
+    }
+    else
+    {
+        ((HybridSolver&) (*fine_solver_)).UpdateAggScaling(coeff);
```

Comment:
I think it is generally recommended to avoid C style casts.

Reply:
Right, I will change it to dynamic_cast, thanks!